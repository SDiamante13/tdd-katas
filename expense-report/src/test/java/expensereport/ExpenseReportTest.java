package expensereport;

import nl.altindag.log.LogCaptor;
import org.approvaltests.ApprovalUtilities;
import org.approvaltests.Approvals;
import org.approvaltests.approvers.ApprovalApprover;
import org.approvaltests.reporters.QuietReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.util.Date;
import java.util.stream.Collectors;

import static java.util.stream.Collectors.joining;

@UseReporter(QuietReporter.class)
class ExpenseReportTest {

    private LogCaptor logCaptor = LogCaptor.forClass(ExpenseReport.class);

    @Test
    void printReport() {
        ExpenseReport expenseReport = new ExpenseReport();

        Expenses expenses = new Expenses(
                new Expense(ExpenseType.DINNER, 100),
                new Expense(ExpenseType.BREAKFAST, 40),
                new Expense(ExpenseType.CAR_RENTAL, 250),
                new Expense(ExpenseType.DINNER, 5001),
                new Expense(ExpenseType.BREAKFAST, 1001)
        );
        expenseReport.printReport(expenses, new Date(0));

        Approvals.verify(getLogs());
    }

    private String getLogs() {
        return logCaptor.getLogs()
                .stream()
                .collect(joining("\n"));
    }
}