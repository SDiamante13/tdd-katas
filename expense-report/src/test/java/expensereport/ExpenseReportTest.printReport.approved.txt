Expenses Wed Dec 31 19:00:00 EST 1969
Dinner	100	 
Breakfast	40	 
Car Rental	250	 
Dinner	5001	X
Breakfast	1001	X
Meal expenses: 6142
Total expenses: 6392
